akka-projection-eventsourced/src/main/scala/akka/projection/eventsourced/scaladsl/EventSourcedProvider.scala

```diff
@@ -273,6 +310,271 @@ object EventSourcedProvider {
 
   }
 
+  private class ReplayableEventsBySlicesSourceProvider[Event](
+      override val system: ActorSystem[_],
+      query: EventsBySliceQuery with CurrentEventsByPersistenceIdTypedQuery,
+      entityType: String,
+      override val minSlice: Int,
+      override val maxSlice: Int,
+      adjustStartOffset: Option[Offset] => Future[Option[Offset]])
+      extends EventsBySlicesSourceProvider[Event](system, query, entityType, minSlice, maxSlice, adjustStartOffset)
+      with CanTriggerReplay
+      with HasGapFillerActor {
+
+    /** INTERNAL API */
+    @InternalApi
+    override private[scaladsl] val gapFillerRef = new AtomicReference()
```

Comment:
Messy with this kinda stateful thing on an interface that is meant to be stateless/without lifecycle. If there was lifecycle for the provider it would be fine to start it on provider start or lazily and then stop on provider stop. :/

Reply:
Or even better, if we could tie it to the specific query